### ISSUE TITLE
xrdfs rm - not exiting with a !=0 exit code in case of errors

### ISSUE THREAD

**Issue by costing** [xrootd/xrootd#2097]
Hi,

Since I can't reopen https://github.com/xrootd/xrootd/issues/1885 I'll copy-paste the request here.

This bug seems to have reappeared at least from 5.6.0 and beyond. Even for single files, regardless of the success or failure of the operation, the exit code is always 0:

```
$ v5.6.0-6/bin/xrdfs eosalice.cern.ch rm '/33/33333/33333'
rm /33/33333/33333 : [ERROR] Error response: permission denied
$ echo $?
0

$ v5.6.0-6/bin/xrdfs eosalice.cern.ch rm '/33/33333/33333' '/44/44444/444444'
rm /33/33333/33333 : [ERROR] Error response: permission denied
rm /44/44444/444444 : [ERROR] Error response: permission denied
$ echo $?
0
```

and the same in 5.6.2:

```
$ xrdcp -version
v5.6.2
$ xrdfs eosalice.cern.ch rm '/33/33333/33333'
rm /33/33333/33333 : [ERROR] Error response: permission denied
$ echo $?
0
```

Could you please re-apply the fix?

Cheers,

.costin

**Comment by smithdh** [comment]
Hello @costing Thanks for the report, sorry for the problem. The sequence of things were is actually a little complicated; the https://github.com/xrootd/xrootd/commit/012c5fe9318ba3ce302be30cb2134dbd3469a5d9 patch helped with the issue of no error return code; and that patch was included in v5.5.2 to 5.5.5. It meant that xrdfs <host> rm <files> would give an error return code if there was an error for each of the files specified to be removed. However that patch caused another problem and I effectively removed it in v5.6.0, as a fix for the other problem, without knowing the impact on this return code. So that returned the situation for xrdfs rm to as it had been with v5.4.2-v5.5.1.

I've made a change now, for the next release, that should result in the intended behaviour, i.e. that xrdfs <host> rm <files> give the error code if there is a problem removing any of the files. (The difference of 'any' files compared to 'all'  is different from what was there between 5.5.2 and 5.5.5). I've also added a test case to our standard tests, to aim to make sure an error return code is always given when expected in future releases.
